homeassistant/components/camera/__init__.py

```diff
@@ -754,7 +731,7 @@ async def async_handle_record_service(camera, call):
     video_path = filename.async_render(variables={ATTR_ENTITY_ID: camera})
 
     data = {
-        CONF_STREAM_SOURCE: source,
+        CONF_STREAM_ID: stream.stream_id,
```

Comment:
Another thought here is to add a method called "record" on the Stream and remove this service entirely, or move "create stream" and "update stream" to be a service rather than a class.